src/Angor/Client/Pages/Spend.razor

```diff
@@ -880,34 +870,57 @@
     {
         var selectedItem = e.Value?.ToString();
 
-        if (selectedItem != null)
+        if (selectedItem != null && int.TryParse(selectedItem, out int res) && res <= feeData.FeeEstimations.Fees.Count)
         {
-            if (int.TryParse(selectedItem, out int res))
-            {
-                if (res <= feeData.FeeEstimations.Fees.Count)
+            feeData.FeePosition = res;
+
+            var selectedFeeEstimation = feeData.FeeEstimations.Fees.OrderBy(fee => fee.Confirmations).ToList()[res - 1];
+
+            await RecalculateFee(selectedFeeEstimation.FeeRate, selectedFeeEstimation.Confirmations);
+        }
+    }
+
+    private async Task RecalculateFee(decimal customFeeRate, int confirmations = 0)
+    {
+        if (!passwordComponent.HasPassword())
+        {
+            showCreateModal = false;
+            notificationComponent.ShowErrorMessage("Wallet password expired");
+            return;
+        }
+
+        try
+        {
+            var words = await passwordComponent.GetWalletAsync();
+
+            var customFeeEstimation = new FeeEstimation
                 {
-                    if (!passwordComponent.HasPassword())
-                    {
-                        showCreateModal = false;
-                        notificationComponent.ShowErrorMessage("Wallet password expired");
-                        return;
-                    }
+                    FeeRate = (int)customFeeRate,
```

Comment:
you are converting a decimal to int where it is represented as a long, it will break the fee estimation

Reply:
You're right, that was a mistake. I've corrected it to ensure the fee estimation works properly. Thanks for the review!